src/DataProviders/SqlDataProvider/SqlDataProvider.ts

```diff
@@ -0,0 +1,68 @@
+﻿//import sql from "mssql";
+const sql = require('mssql');
+import * as assert from "assert";
+import { IDataProvider } from "../IDataProvider";
+import { ConnectionPool, ConnectionError, config, IResult } from "mssql";
+import { SeasonInfo } from "../../DataModels/SeasonInfo";
+
+/*
+ * Sql Data Provider
+ */
+export class SqlDataProvider implements IDataProvider {
+    private _tablePrefix: string = "Hymns_";
+
+    constructor(private _config: config) {
+    }
+
+    private _getQualifiedName(sp: string): string {
+        return `${this._tablePrefix}${sp}`
+    }
+
+    //TODO: why is the return type requiring void?
```

Comment:
probably because you have a `catch` that doesn't return a `SeasonInfo[]` or throw another exception, but returns an empty promise. I would suggest you either:
a) get rid of the `catch` from here completely and handle this one layer up, or
b) re-throw the error after you do the logging in your current `catch`.